### PR TITLE
NR-103060 Solidify telemetry in the Observability panel

### PR DESCRIPTION
**Pull request by dsellarsnr**
**This PR Addresses:**  
[NR-103060 Solidify telemetry in the Observability panel](https://issues.newrelic.com/browse/NR-103060)  

<sup> Created from JetBrains using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)</sup>

### CODE REVIEW

**Comment by bcanzanella** [comment]
@dsellarsnr lots of gnarly stuff here, do u think some React tests would help ensure everything is ok? i'm sure there would be tons of mocking + spies...

File: shared/ui/Stream/Observability.tsx [hunk 1]
```diff
@@ -522,40 +513,43 @@ export const Observability = React.memo((props: Props) => {
...
-				HostApi.instance.track("O11y Rendered", {
-					State: telemetryStateValue,
-				});
-			}
+		// "Not Connected" - not connected to NR, this goes away with UID completion
+		if (!derivedState.newRelicIsConnected) {
+			telemetryStateValue = "Not Connected";
+		}
 
-			if (expandedEntity) {
```
[the remark this answers is not in the document]
Comment (dsellarsnr) [reply shared/ui/Stream/Observability.tsx:552]: Yes, it relies on data that isn't available at the time this render telemetry event fires so I have each separate telemetry call wait on their own data dependencies.

**Comment by dsellarsnr** [comment]
> @dsellarsnr lots of gnarly stuff here, do u think some React tests would help ensure everything is ok? i'm sure there would be tons of mocking + spies...

yeah, it is gnarly. do you want it as a separate ticket? without this change it is definitely going to have some wrong metrics. but the tests will take it into next week.

**Comment by bcanzanella** [comment]
> > @dsellarsnr lots of gnarly stuff here, do u think some React tests would help ensure everything is ok? i'm sure there would be tons of mocking + spies...
> 
> yeah, it is gnarly. do you want it as a separate ticket? without this change it is definitely going to have some wrong metrics. but the tests will take it into next week.

@dsellarsnr let's get this in then tack on the tests after. thanks.

**Review by bcanzanella — APPROVED** [review]

File: shared/ui/Stream/Observability.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dsellarsnr) [reply shared/ui/Stream/Observability.tsx:552]: <ObservabilityCurrentRepo> -> setCurrentRepoId ->  useEffect on [currentRepoId, observabilityRepos] calls setExpandedEntity -> triggers useEffect dedicated to callServiceClickedTelemetry()

[the remark this answers is not in the document]
Comment (bcanzanella) [reply shared/ui/Stream/Observability.tsx:552]: you forgot the `/s` :=)

**Pull request merged**